### PR TITLE
Allow specific format width

### PR DESCRIPTION
**Pull request by xsebek**
* add option `--width` to `format` CLI
* if width is not specified use terminal width
* if terminal width is unknown, use default (100)

This should help with testing (not so) long layouts like in #1473.

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -157,16 +164,24 @@ showInput :: Input -> Text
 showInput Stdin = "(input)"
 showInput (File fp) = pack fp
 
+infixl 3 <??>
+(<??>) :: Maybe a -> a -> a
+(<??>) = flip fromMaybe
```
Comment (byorgey) [inline app/Main.hs:169]: Note, we already have this defined in `Swarm.Util` as `(?)`. (But `infixr 1` instead of `infixl 3`.)

File: app/Main.hs [hunk 2]
```diff
@@ -157,16 +164,24 @@ showInput :: Input -> Text
...
-        Just ast -> Text.putStrLn $ prettyText ast
-      exitSuccess
+    Right Nothing -> Text.putStrLn ""
+    Right (Just ast) -> do
+      mWindow <- Term.size
+      let mkOpt w = LayoutOptions (AvailablePerLine w 1.0)
+      let opt =
+            fmap mkOpt mWidth
+              <|> fmap (\(Term.Window _h w) -> mkOpt w) mWindow
+                <??> defaultLayoutOptions
```
Comment (byorgey) [inline app/Main.hs:183]: If you use an `infixr` version of `<??>` / `?` you can just write this as a chain (untested):
```
fmap mkOpt mWidth
  ? fmap (\(Term.Window _h w) -> mkOpt w) mWindow
  ? defaultLayoutOptions
```

**Review by byorgey — APPROVED** [review]

File: app/Main.hs (hunk 2 above)
Comment (xsebek) [reply app/Main.hs]: Works great thanks!

File: app/Main.hs (hunk 1 above)
Comment (xsebek) [reply app/Main.hs]: Ah, I saw Relude has [`(?:)`](https://hackage.haskell.org/package/relude-1.2.0.0/docs/Relude-Monad-Maybe.html#v:-63-:) as `infixr 0` but I did not know we have some operator too, thanks! 👍

**Pull request merged**